code/modules/hydroponics/hydroponics.dm

```diff
@@ -42,6 +42,14 @@
 	icon = 'icons/obj/hydroponics/equipment.dmi'
 	icon_state = "hydrotray3"
 
+/obj/machinery/hydroponics/constructable/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/simple_rotation, ROTATION_ALTCLICK | ROTATION_CLOCKWISE | ROTATION_COUNTERCLOCKWISE | ROTATION_VERBS, null, CALLBACK(src, .proc/can_be_rotated))
+	AddComponent(/datum/component/plumbing/simple_demand)
```

Comment:
this needs to be in `Initialize()` after `create_reagents` and not `ComponentInitialize()` or else it runtimes because it tries to add the component before reagents are initialized (and the component doesn't attach either, since there's no reagent capacity to work with)